src/encoded/static/components/filegallery.js

```diff
@@ -1986,8 +1986,154 @@ function createFacetObject(propertyKey, fileList, filters) {
+            // that lab. Also form the lab title here, prepending with the rfa (e.g. ENCODE3) for
+            // `UNIFORM_PIPELINE_LAB`.
+            const analysesByLab = _(qualifyingAnalyses).groupBy((analyses) => {
+                if (analyses.pipeline_labs.length > 1) {
+                    return 'Mixed';
+                }
+
+                // At this stage, we know analyses.pipeline_labs has one and only one element.
+                if (analyses.pipeline_labs[0] !== UNIFORM_PIPELINE_LAB) {
+                    return 'Lab custom';
+                }
+                return `${analyses.pipeline_award_rfas[0]} uniform`;
+            });
+
+            // Fill in the compiled object with the labs that group the files.
+            compiledAnalyses = [];
+            const fileIds = files.map(file => file['@id']);
+            Object.keys(analysesByLab).forEach((pipelineLab) => {
+                // For one lab, group all analyses by their assembly/annotation, then combine all
+                // file arrays for those with matching pipeline lab, assembly/annotation, and RFA.
+                const analysesByAssembly = _(analysesByLab[pipelineLab]).groupBy(analyses => `${analyses.assemblies[0]}${analyses.genome_annotations.length === 1 ? ` ${analyses.genome_annotations[0]}` : ''}`);
+                Object.keys(analysesByAssembly).forEach((assembly) => {
+                    // Combine all analyses files that share the same pipeline lab, assembly, and
+                    // annotation and add each to the compiled list. Filter out any not included in
+                    // the experiment's files.
+                    const assemblyFiles = _.uniq(analysesByAssembly[assembly].reduce((accFiles, analyses) => accFiles.concat(analyses.files), []).filter(file => fileIds.includes(file)));
+                    if (assemblyFiles.length > 0) {
+                        compiledAnalyses.push({
+                            title: `${pipelineLab} ${assembly}`,
+                            pipelineLab,
+                            assembly,
+                            assemblyAnnotationValue: computeAssemblyAnnotationValue(analysesByAssembly[assembly][0].assemblies[0], analysesByAssembly[assembly][0].genome_annotations[0]),
+                            files: _.uniq(assemblyFiles),
+                        });
+                    }
+                });
+            });
+        }
+    }
+    return _(compiledAnalyses).sortBy(compiledAnalysis => -compiledAnalysis.assemblyAnnotationValue);
+};
+
+
+/**
+ * Find the compiled analyses object matching the given assembly and annotation and return its
+ * index in the compiled analyses array.
+ * match found.
+ * @param {string} assembly Assembly and annotation to map to an analyses title
+ * @param {array} compiledAnalyses Compiled analyses for the experiment
+ *
+ * @return {object} Array index of compiled analyses object matching given assembly, or -1
```

Comment:
Yeah, JSDoc likes only a space after the variable name, but I’m not sure what I think about JSDoc anymore. I complied with it so it’ll pass JSDoc tools, but we never run those tools because we have so much code, even new code, that doesn’t comply, and it works poorly for React functional components. It might be best just to do whatever seems clearest, going forward.